artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java

```diff
@@ -1564,6 +1562,7 @@ private void clearRedistributorFuture() {
    @Override
    public synchronized void cancelRedistributor() throws Exception {
       if (redistributor != null) {
+         consumers.remove(redistributor);
```

Comment:
The redistributor.consumer.stop() below this removal can throw, which would prevent the "redistributor = null" below that. Now that the getConsumerSize() return  is contingent on both the consumers map size and the redistributor nulling, perhaps this potential for them to get out of alignment should be handled now?

Reply:
good point and good catch, it should be handled. thanks!

Reply:
sorted, but going to tidy up the catch of that potential throw... it is not consisten

Reply:
should be sorted now, thanks for the second set of eyes. turns out the log message was a cut and paste!